geopandas/tests/test_crs.py

```diff
@@ -128,3 +132,381 @@ def test_skip_exact_same():
+
+    @pytest.mark.filterwarnings("ignore:Assigning CRS")
+    def test_dataframe(self):
+        arr = from_shapely(self.geoms, crs=27700)
+        s = GeoSeries(arr)
+        df = GeoDataFrame(geometry=s)
+        assert df.crs == self.osgb
+        assert df.geometry.crs == self.osgb
+        assert df.geometry.values.crs == self.osgb
+
+        arr = from_shapely(self.geoms)
+        s = GeoSeries(arr, crs=27700)
+        df = GeoDataFrame(geometry=s)
+        assert df.crs == self.osgb
+        assert df.geometry.crs == self.osgb
+        assert df.geometry.values.crs == self.osgb
+
+        # different passed CRS than array CRS is ignored
+        df = GeoDataFrame(geometry=s, crs=4326)
+        assert df.crs == self.osgb
+        assert df.geometry.crs == self.osgb
+        assert df.geometry.values.crs == self.osgb
+        with pytest.warns(FutureWarning):
+            GeoDataFrame(geometry=s, crs=4326)
+        with pytest.warns(FutureWarning):
+            GeoDataFrame({"data": [1, 2], "geometry": s}, crs=4326)
+        with pytest.warns(FutureWarning):
+            GeoDataFrame(df, crs=27700).crs
+
+        # manually change CRS
+        df = GeoDataFrame(geometry=s)
+        df.crs = 4326
+        assert df.crs == self.wgs
+        assert df.geometry.crs == self.wgs
+        assert df.geometry.values.crs == self.wgs
+
+        df = GeoDataFrame(self.geoms, columns=["geom"], crs=27700)
+        assert df.crs == self.osgb
+        df = df.set_geometry("geom")
+        assert df.crs == self.osgb
+        assert df.geometry.crs == self.osgb
+        assert df.geometry.values.crs == self.osgb
+        assert df.geom.crs == self.osgb
+        assert df.geom.values.crs == self.osgb
+
+        df = GeoDataFrame(geometry=self.geoms, crs=27700)
+        assert df.crs == self.osgb
+        assert df.geometry.crs == self.osgb
+        assert df.geometry.values.crs == self.osgb
+
+        df = GeoDataFrame(crs=27700)
+        df = df.set_geometry(self.geoms)
+        assert df.crs == self.osgb
+        assert df.geometry.crs == self.osgb
+        assert df.geometry.values.crs == self.osgb
+
+        # new geometry with set CRS has priority over GDF CRS
+        df = GeoDataFrame(crs=27700)
+        df = df.set_geometry(self.geoms, crs=4326)
+        assert df.crs == self.wgs
+        assert df.geometry.crs == self.wgs
+        assert df.geometry.values.crs == self.wgs
+
+        df = GeoDataFrame()
+        df = df.set_geometry(s)
+        assert df.crs == self.osgb
+        assert df.geometry.crs == self.osgb
+        assert df.geometry.values.crs == self.osgb
+
+        arr = from_shapely(self.geoms, crs=27700)
+        df = GeoDataFrame()
+        df = df.set_geometry(arr)
+        assert df.crs == self.osgb
+        assert df.geometry.crs == self.osgb
+        assert df.geometry.values.crs == self.osgb
+
+        arr = from_shapely(self.geoms)
+        df = GeoDataFrame({"col1": [1, 2], "geometry": arr}, crs=4326)
```

Comment:
Is the case where geometry is inside the dict (so not with `geometry=` keyword) and has a crs covered?

(above in this test function it seems mainly geometry= pattern)

Reply:
Added test for that.